### ISSUE TITLE
Loop inside Loop finnishes abruptly - v1.29.1

### ISSUE THREAD

**Issue by cristian-dan-f** [n8n-io/n8n#8764]
### Bug Description

I have been looking at this for hours. I am using N8N online version v1.29.1

I have a Loop inside another Loop and the problem is that the second Loop only runs once. For example if there are 4 items, it will loop 4 times but only once on the 'Loop Branch', the other 3 goes to 'Done Branch'. So it recognizes the items, but for some reason it takes data from a Node inside the Loop, so it is mixing data.

What i do is simple. I am creating Terms for different Taxonomies for WordPress. So I have an initial Loop for Taxonomies, and then a second loop for each Term not present in WP. I attach a screenshot.

Both Loop data is perfect, I just can't make it create more then 1 Term in each execution.
Have a look at this JSON, it is the Output of the Loop Node, you will instantly see the problem:

```
[
	{
		"new_terms": {
			"time_prepare": [
				{
					"term": 634
				}
			]
		}
	},
	{
		"term": "Almuerzos vegetarianos"
	},
	{
		"term": "Comida balanceada"
	},
	{
		"term": "fusilli"
	},
	{
		"term": "vegetariano"
	},
	{
		"term": "hervido"
	},
	{
		"term": "horneado"
	},
	{
		"term": "España"
	},
	{
		"term": "recetas fáciles almuerzo"
	}
]
```

The first Object in the Array, that contains 'new_terms' should NOT be there. It comes from a  Code Node (named Store - WP Term) inside the second loop right at the end of the loop.

![n8n-loop-inside-loop-finnishes-abruptly](https://github.com/n8n-io/n8n/assets/100019968/7aee08e7-02c5-45a3-8c4c-74fb308b15d5)

### To Reproduce

-

### Expected behavior

-

### Operating System

online version

### n8n Version

1.29.1

### Node.js Version

online version

### Database

SQLite (default)

### Execution mode

main (default)

**Comment by Joffcom** [comment]
Hey @cristian-dan-f,

Can you provide a minimal workflow that reproduces the issue?

**Comment by cristian-dan-f** [comment]
@Joffcom I hope I have done it right. I attach a test workflow.
It has example data, and it should execute without problems.

[Bug_Loop_Inside_Loop.json](https://github.com/n8n-io/n8n/files/14437917/Bug_Loop_Inside_Loop.json)

**Comment by Joffcom** [comment]
Hey @cristian-dan-f,

Have you tried resetting your loop or using a sub workflow instead? We don't recommend using nested loops as it can cause issues if you don't reset the loop correctly so instead we would recommend moving the second loop to a sub workflow and using that instead.

**Comment by cristian-dan-f** [comment]
I tried to check the Reset but it fails on second loop. I finally managed to do one loop inside a Code Node. The end result is very satisfactory, although I am moving away from the 'no-code' idea I had in the beginning for N8N

**Comment by Joffcom** [comment]
The good news is we are looking at changing the loop node to handle multiple executions better which will be a massive help for cases like this.

**Comment by Joffcom** [comment]
We’re taking some time to review and tidy up our issue tracker. It looks like this issue hasn’t had any recent activity for the past 6 months. To keep our backlog focused on current priorities, we’re closing this issue for now.

That said, this doesn’t mean the issue is forgotten! If this is still relevant or you have more information to add, feel free to reopen it or create a new issue and we’d be happy to revisit it.

Thank you for understanding and for contributing to the project!
